django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -64,92 +61,110 @@ class PackageListResponseSerializer(serializers.Serializer):
     results = CyberstormPackagePreviewSerializer(many=True)
 
 
-class BasePackageListApiView(ListAPIView):
+class PackageListPaginator(PageNumberPagination):
+    page_size = 20
+
+    def get_schema_fields(self, view):
+        """
+        By default this would return page (via page_query_param
+        attribute) and page_size (via page_size_query_param attribute).
+        The former would clash with the page field defined in our
+        PackageListRequestSerializer, breaking the API documentation
+        generated by drf_yasg. Returning an empty list means we have the
+        control and responsibility to make sure the documentation
+        matches reality.
+        """
+        return []
+
+
+class BasePackageListAPIView(ListAPIView):
     """
     Base class for community-scoped, paginated, filterable package listings.
 
     Classes implementing this base class should receive `community_id`
-    url parameter and implement the abstract methods listed below.
+    url parameter.
+
+    Methods with names starting prefixed with underscore are you custom
+    methods, whereas the rest are overwritten methods from ListAPIView.
     """
 
-    page_size = 20
+    pagination_class = PackageListPaginator
+    serializer_class = CyberstormPackagePreviewSerializer
+    viewname: str = ""  # Define in subclass
 
-    @abstractmethod
-    def _get_paginator_cache_key(self) -> str:
-        """
-        Return cache_key argument for CachedPaginator initialization.
-        """
-        raise NotImplementedError(
-            "PackageListApiView must implement _get_paginator_cache_key",
-        )
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
 
-    @abstractmethod
-    def _get_paginator_cache_vary_prefix(self) -> str:
-        """
-        Return endpoint and URL parameter specific string.
+    def list(self, request, *args, **kwargs):  # noqa: A003
+        assert self.paginator is not None
```

Comment:
Asserts should not be used outside of tests as they might be optimized away in some python runtimes. The preferred approach is to raise an actual error yourself.

In practice this shouldn't be an issue for us at least right now, and I doubt we'd swap to a runtime which ignores them without making sure it doesn't lead to anything unexpected first, but still figured I'd mention it.

Not a blocker

Reply:
> Asserts should not be used outside of tests as they might be optimized away in some python runtimes.

Alright, did not know this. FWIW I used asserts as DRF implementation itself uses them, e.g. [to check the self.paginator value](https://www.cdrf.co/3.14/rest_framework.generics/ListAPIView.html#get_paginated_response).

Reply:
Looks like they didn't either 😄 but really I'd blame python for making that a thing in the first place (just running with `python -O` will skip asserts)

Reply:
(wontfix)